src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -2,58 +2,241 @@
 // Licensed under the MIT License.
 using System.Threading;
 using System.Threading.Tasks;
+using Bicep.Core.FileSystem;
 using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Completions;
 using Bicep.LanguageServer.Extensions;
 using Bicep.LanguageServer.Providers;
 using Bicep.LanguageServer.Utils;
 using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
 using OmniSharp.Extensions.LanguageServer.Protocol.Document;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using System;
+using Bicep.Core.Parsing;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using System.Linq;
 
 namespace Bicep.LanguageServer.Handlers
 {
     public class BicepDefinitionHandler : DefinitionHandlerBase
     {
         private readonly ISymbolResolver symbolResolver;
 
-        public BicepDefinitionHandler(ISymbolResolver symbolResolver) : base()
+        private readonly ICompilationManager compilationManager;
+
+        private readonly IFileResolver fileResolver;
+
+        public BicepDefinitionHandler(ISymbolResolver symbolResolver, ICompilationManager compilationManager, IFileResolver fileResolver) : base()
         {
             this.symbolResolver = symbolResolver;
+            this.compilationManager = compilationManager;
+            this.fileResolver = fileResolver;
         }
 
         public override Task<LocationOrLocationLinks> Handle(DefinitionParams request, CancellationToken cancellationToken)
         {
+            var context = this.compilationManager.GetCompilation(request.TextDocument.Uri);
+            if (context == null)
+            {
+                return Task.FromResult(new LocationOrLocationLinks());
+            }
             var result = this.symbolResolver.ResolveSymbol(request.TextDocument.Uri, request.Position);
             if (result == null)
             {
+                // Currently we only definition handler for a non symbol bound to implement module path goto.
+                // try to resolve module path syntax from given offset using syntax matching.
+                int offset = PositionHelper.GetOffset(context.LineStarts, request.Position);
+                var matchingNodes = SyntaxMatcher.FindNodesMatchingOffset(context.Compilation.SourceFileGrouping.EntryPoint.ProgramSyntax, offset);
+                if (SyntaxMatcher.IsTailMatch<ModuleDeclarationSyntax, StringSyntax, Token>(
```

Comment:
>SyntaxMatcher [](http://example.com/codeflow?start=20&length=13)

Indentation is weird here. I'd recommend something like this:
`doThing(
    firstArg,
    secondArg);

I think this way the predicate can be on one line.